ksp_plugin/orbit_analyser.cpp

```diff
@@ -31,21 +31,22 @@ OrbitAnalyser::OrbitAnalyser(
       analysed_trajectory_parameters_(
           std::move(analysed_trajectory_parameters)) {}
 
-void OrbitAnalyser::Restart() {
-  analyser_ = MakeStoppableThread([this] { RepeatedlyAnalyseOrbit(); });
+void OrbitAnalyser::Interrupt() {
+  analyser_ = jthread();
```

Comment:
I would `request_stop` and `detach` (only if the thread is joinable, of course).  The thread would immediately become non-joinable, and could happily execute for a while until it notices the stop.  You could then drop the `analyzer_done_` boolean since it would effectively be subsumed by `joinable`.

Reply:
Rewrote as discussed.